libs/datamodel/core/tests/reformat/reformat.rs

```diff
@@ -74,6 +74,146 @@ model Post {
     assert_reformat(input, expected);
 }
 
+#[test]
+fn format_should_enforce_order_of_field_directives() {
+    let input = r#"model Post {
+  id        Int      @default(autoincrement()) @id
+  content   String?
+  published Boolean  @default(false) @map("_published")
+  author    User?   @relation(fields: [authorId], references: [id])
+  authorId  Int?
+}
+"#;
+    let expected = r#"model Post {
+  id        Int     @id @default(autoincrement())
+  content   String?
+  published Boolean @map("_published") @default(false)
+  author    User?   @relation(fields: [authorId], references: [id])
+  authorId  Int?
+}
+"#;
+
+    assert_reformat(input, expected);
+}
+
+#[test]
+fn format_should_enforce_order_of_block_directives() {
```

Comment:
It might be good to have a model that uses all block level directives to ensure the order is correct.